lib/routes/18comic/search.ts

```diff
@@ -40,11 +44,56 @@ async function handler(ctx) {
     const category = ctx.req.param('category') ?? 'all';
     const keyword = ctx.req.param('keyword') ?? '';
     const time = ctx.req.param('time') ?? 'a';
-    const order = ctx.req.param('order') ?? 'mr';
     const { domain = defaultDomain } = ctx.req.query();
     const rootUrl = getRootUrl(domain);
-
+    let order = ctx.req.param('order') ?? 'mr';
     const currentUrl = `${rootUrl}/search/${option}${category === 'all' ? '' : `/${category}`}${keyword ? `?search_query=${keyword}` : '?'}${time === 'a' ? '' : `&t=${time}`}${order === 'mr' ? '' : `&o=${order}`}`;
+    const limit = ctx.req.query('limit') ? Number.parseInt(ctx.req.query('limit')) : 20;
+
+    let apiUrl = getApiUrl();
+    order = time === 'a' ? order : `${order}_${time}`;
+    apiUrl = `${apiUrl}/search?search_query=${keyword}&o=${order}`;
+    let apiResult = await processApiItems(apiUrl);
+    let filteredItemsByCategory = apiResult.content;
+    // Filter items by category if not 'all'
+    if (category !== 'all') {
+        filteredItemsByCategory = apiResult.content.filter((item) => item.category.title === apiMapCategory(category));
+    }
+    filteredItemsByCategory = filteredItemsByCategory.slice(0, limit);
+    const results = await Promise.all(
+        filteredItemsByCategory.map((item) =>
+            cache.tryGet(item.id, async () => {
```

Comment:
Don't simply use `id` as the cache key. It's way too easy for a cache collision to occur.